sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/state/ViewportState.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import com.mapbox.maps.plugin.viewport.ViewportPlugin
+import com.mapbox.maps.plugin.viewport.experimental.data.ViewportDataSource
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportStateTransition
+
+/**
+ * Set of possible [ViewportPlugin] states.
+ */
+interface ViewportState {
+  val id: String
+  val isTransitioning: Boolean
+  val dataSource: ViewportDataSource?
+  val stateTransition: ViewportStateTransition?
+
+  object IDLE : ViewportState {
```

Comment:
I'd consider modeling this as a null current state.

Reply:
sounds good to me!